akka-stream/src/main/scala/akka/stream/scaladsl/Hub.scala

```diff
@@ -987,24 +980,36 @@ private[akka] class PartitionHub[T](
 
     private val queue = createQueue()
     private var pending = Vector.empty[T]
-    private var consumerInfo: ConsumerInfoImpl = new ConsumerInfoImpl(Array.emptyLongArray, Vector.empty)
+    private var consumerInfo: ConsumerInfoImpl = new ConsumerInfoImpl(Vector.empty)
     private val needWakeup: LongMap[Consumer] = LongMap.empty
 
     private var callbackCount = 0L
 
-    private class ConsumerInfoImpl(override val consumerIds: Array[Long], val consumers: Vector[Consumer])
-      extends ConsumerInfo {
+    private final class ConsumerInfoImpl(val consumers: Vector[Consumer])
+      extends ConsumerInfo { info ⇒
 
       override def queueSize(consumerId: Long): Int =
         queue.size(consumerId)
 
-      override def size: Int = consumerIds.length
+      override def size: Int = consumers.length
+
+      override def consumerIds: immutable.IndexedSeq[Long] =
+        consumers.map(_.id)
+
+      override def consumerIdByIdx(idx: Int): Long =
+        consumers(idx).id
+
+      override def getConsumerIds: java.util.List[Long] =
+        new util.AbstractList[Long] {
```

Comment:
`JavaConverters.seqAsJavaList` seems to do basically the same but this one is fine for me as well.